src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -2468,18 +2468,27 @@ private function registerHttpClientConfiguration(array $config, ContainerBuilder
             $retryOptions = $scopeConfig['retry_failed'] ?? ['enabled' => false];
             unset($scopeConfig['retry_failed']);
 
+            $httpClientTransport = new Reference('http_client.transport');
+
+            if ($responseFactoryId = $scopeConfig['mock_response_factory'] ?? null) {
+                $container->register('http_client.mock_client.'.$name, MockHttpClient::class)
+                    ->setDecoratedService($httpClientTransport, null, -10)  // lower priority than TraceableHttpClient (5)
```

Comment:
it is also decorated with priority `10` for the global `mock_response_factory`. Maybe worth having a lower priority here so it superseeds the global one ?